Задание 2-1

ThewayofthesmuraiZero opened the pull request:
ЗАДАНИЕ 2-1
ВАРИАНТ 11
![image](https://user-images.githubusercontent.com/70698521/113870701-c92fdb00-97ba-11eb-930f-1d75ceeb8739.png)

Создать консольное приложение для решения задачи, представленной
в таблице, с использованием перечислимого типа. Выбор формулы вычисления
 зависит от пользователя. Данные для решения задачи так же вводит
пользователь. Ввод необходимо проверять на правильность.
Все результаты вывести на экран. Дополнить свой отчёт блок-схемой
алгоритма.

ЗАДАНИЕ 2-2
ВАРИАНТ 11
![image](https://user-images.githubusercontent.com/70698521/113870742-d51b9d00-97ba-11eb-8be8-b0d6803dce99.png)

Создать консольное приложение с организацией разветвлённого процесса
для решения заданной в таблице функции. Данные для решения вводит
пользователь. Ввод необходимо проверять на правильность. Все результаты
 вывести на экран. Дополнить свой отчёт блок-схемой алгоритма.

РЕШЕНИЕ В EXEL 
![image](https://user-images.githubusercontent.com/70698521/113875348-55440180-97bf-11eb-8247-eb8b27efbb77.png)
![image](https://user-images.githubusercontent.com/70698521/113875411-63921d80-97bf-11eb-87f9-0bacaa1a33fe.png)

Задание 2-3 C#
Создать консольное приложение с организацией разветвлённого процесса для решения задачи из таблицы. Данные для решения вводит пользователь. Ввод необходимо проверять на правильность. Все результаты вывести на экран. Дополнить свой отчёт блок-схемой алгоритма.
ВАРИАНТ 11
![image](https://user-images.githubusercontent.com/70698521/114061948-a297a000-989f-11eb-8ffd-f9db25918a6c.png)


Задание 2-3 MatLab

Построить график функции, заданной в таблице, и дать ему название. В качестве данных для решения использовать константы. Пользовательский ввод можно не использовать. Все результаты вывести на экран.
![image](https://user-images.githubusercontent.com/70698521/114062009-b3481600-989f-11eb-8659-0add247ab6e7.png)

ThewayofthesmuraiZero committed:
Задание 2-1

ВАРИАНТ 11
Создать консольное приложение для решения задачи, представленной
в таблице, с использованием перечислимого типа. Выбор формулы вычисления
 зависит от пользователя. Данные для решения задачи так же вводит
пользователь. Ввод необходимо проверять на правильность.
Все результаты вывести на экран. Дополнить свой отчёт блок-схемой
алгоритма.

ЗАДАНИЕ 2-2
ВАРИАНТ 11
Создать консольное приложение с организацией разветвлённого процесса
для решения заданной в таблице функции. Данные для решения вводит
пользователь. Ввод необходимо проверять на правильность. Все результаты
 вывести на экран. Дополнить свой отчёт блок-схемой алгоритма.

ThewayofthesmuraiZero committed:
Update Task 2-2.cpp

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,40 @@
+﻿
+#include <iostream>
+using namespace std;
+ 
```
Добавить перечислния 
```cpp
class enum Action 
{ volume = 1, FullFaceArea = 2, faceArea = 3};

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,40 @@
+    double a, v, sp, sg;
+    int b;
+
+    cout << " Введите грань: \n";
```
```cpp
cout << "Введите длину ребра";
double length;
cin >> length;

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,40 @@
+    cout << " 1 - Объем \n";
+    cout << " 2 - Площадь полной поверхности \n";
+    cout << " 3 - Площадь грани \n";
+
```
`int userIntChoice;`

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,40 @@
+    cout << " 3 - Площадь грани \n";
+
+    cin >> b;
+
```
преобразовать тип `int` к типу `enum` в данном случае тип называется `Action`
`const auto userChoice = static_cast<Action>(userIntChoice);`

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,40 @@
+    switch (b)
+    {
+        case 1:
+            v = a * a * a;
```
`const auto volume = getVolume(length);

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,31 @@
+
+    setlocale(LC_ALL, "RUSSIAN");
+    
+    const double a = 0.3;
```
`const auto a = 0.3;`

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,31 @@
+    double x;
+    cout << "Введите переменную х = ";
+    cin >> x;
+    double result;
```
не объявляется заранее

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,31 @@
+
+    if ((x * a) < 1)
+    {
+        result = a* sin((x * x - 1) / 10);
```
`const auto result = getFunc1(a, x);`

MarinaPaley reviewed (changes requested):
Исправить, согласно комментариям.
Выделять функции.
Добавлять документацию.

ThewayofthesmuraiZero committed:
Update Task 2-1

Внесла изменения и документацию

ThewayofthesmuraiZero committed:
Update Task 2-2

Внесла изменения по вашим рекомендациям.

ThewayofthesmuraiZero committed:
ЗАДАЧА 2-3

Создать консольное приложение с организацией разветвлённого процесса
для решения задачи из таблицы. Данные для решения вводит пользователь.
Ввод необходимо проверять на правильность.
Все результаты вывести на экран. Дополнить свой отчёт блок-схемой
алгоритма.

ВАРИАНТ 11
Даны три положительных числа. Определить, можно ли построить
треугольник с длинами сторон, равным этим числам.
Если можно, то ответить на вопрос, является ли он остроугольным

Я проверяю 2 условия:
1) Треугольник строится или нет
2) Треугольник является остроугольным

ThewayofthesmuraiZero committed:
Merge branch 'Feature/2' of https://github.com/ThewayofthesmuraiZero/C---programming into Feature/2

MarinaPaley commented on TASK 2-3.cpp:
```diff
@@ -0,0 +1,53 @@
+ * \param c целое число
+ * \return true при условии, что треугольник возможно построить, после является ли он остроугольным
+ */
+bool TriangleRules(const int a, const int b, const int c);
```
не самое лучшее название `isTriangleExists`
точно неправильно - это тип входных переменных: должно быть вещественное число

MarinaPaley commented on TASK 2-3.cpp:
```diff
@@ -0,0 +1,53 @@
+/**
+ * \brief
+ * Функция определения является ли треугольник остроугольным?
+ * \param a целое число
```
сторона?

MarinaPaley commented on TASK 2-3.cpp:
```diff
@@ -0,0 +1,53 @@
+ * \param a целое число
+ * \param b целое число
+ * \param c целое число
+ * \return true при условии, что треугольник возможно построить, после является ли он остроугольным
```
документация нужна к каждому методу своя

MarinaPaley commented on TASK 2-3.cpp:
```diff
@@ -0,0 +1,53 @@
+ * \return true при условии, что треугольник возможно построить, после является ли он остроугольным
+ */
+bool TriangleRules(const int a, const int b, const int c);
+bool OstrTreangle(const int a, const int b, const int c);
```
`isAcuteTriangle`

MarinaPaley commented on TASK 2-3.cpp:
```diff
@@ -0,0 +1,53 @@
+        ? "Треугольник возможно построить"
+        : "Треугольник нельзя построить ";
+    
+    const auto result2 = OstrTreangle(a, b, c)
```
второй расчет можно делать только втом случае, если треугольник существует
"Если можно, то ответить на вопрос, является ли он остроугольным"

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,80 @@
+/// длина ребракуба
+///найти по выбору пользователя - 1 площадь грани 2 площадь полной поверхности 3 объем
+
+double Facearea(const double cubeedge);
```
Для каждого метода своя документация
Соблюдайте стиль наиманования методов и переменных `FaceArea`, `cubeEdge`

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,80 @@
+ * \cubeedge - ребро куба
+ */
+
+enum ActionChoice { facearea, totalSurfaceArea, volumeCube };
```
`faceArea`

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,41 @@
+
+using namespace std;
+
+double f1(const double x, const double a);
```
Документация

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,41 @@
+
+    setlocale(LC_ALL, "RUSSIAN");
+    
+       const auto a = 0.3;
```
зачем отступ?

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,41 @@
+    double x;
+    cout << "Введите переменную x = ";
+    cin >> x;
+    if (x*a< 1)
```
для повышения читаемости, оставляйте пробелы `x * a < 1`

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,41 @@
+
+double f2(const double x, const double a)
+{
+    returna* cos((x - 1) / 10);
```
пробел

MarinaPaley reviewed (changes requested):
Каждая задача  - в своей ветке. Теперь ждите, когла все задачи будут правильные

ThewayofthesmuraiZero committed:
UP Task 2-2

ThewayofthesmuraiZero committed:
Update Task 2-2.cpp

ThewayofthesmuraiZero committed:
UP Task 2-1

ThewayofthesmuraiZero committed:
Update Task 2-2.cpp

ThewayofthesmuraiZero committed:
Update Task 2-2.cpp

ThewayofthesmuraiZero committed:
Update Task 2-3

Задача 2-3 внесла изменения.

MarinaPaley commented on TASK 2-3.cpp:
```diff
@@ -0,0 +1,42 @@
+ * \param c - вторая сторона стреугольника
+ * \return true при условии, что треугольник возможно построить, после ответить ли он остроугольным?
+ */
+bool isTriangleExists(const double a, const double b, const double c);
```
Так нельзя. У Вас точно две функции. Они должны быть независимыми. Одна- `ITriangleExist`, вторая - (я уже переводила) - Треугольник Остроугольный

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,84 @@
+ * \cubeedge - ребро куба
+ */
+
+double FaceArea(const double cubeedge);
```
Лучше `GetFaceArea`
 Это не документация((( скопируйте у меня
У Вас задача все время про куб. Можно не усложнять - `edge` понятно

MarinaPaley commented on Task 2-1.cpp:
```diff
@@ -0,0 +1,84 @@
+
+
+
+enum ActionChoice { FaceFrea, TotalSurfaceArea, VolumeCube };
```
Лучше `enun class...`

MarinaPaley commented on Task 2-2.cpp:
```diff
@@ -0,0 +1,54 @@
+        cout << "y = " << f2(x, a);
+    }
+    
+    return(0);
```
Зачем скобки?

MarinaPaley reviewed (changes requested):
Каждая задача должна быть в отдельной ветке и, соответственно, отдельном pull request.
Иначе, пока, все задачи не будут правильно решены, не будет approve

ThewayofthesmuraiZero committed:
Update TASK 2-3.cpp
